app/standalone.go

```diff
@@ -120,6 +122,32 @@ func doStandaloneInstallStatesDownload(art io.ReadCloser, key []byte,
 	}
 
 	standaloneData.artifactName = installer.GetArtifactName()
+	standaloneData.artifactTypeProvides, err = installer.GetArtifactProvides()
+	if err != nil {
+		return nil, err
+	}
+	if standaloneData.artifactTypeProvides != nil {
+		if _, ok := standaloneData.artifactTypeProvides["artifact_name"]; ok {
+			delete(standaloneData.artifactTypeProvides, "artifact_name")
+		}
+		if grp, ok := standaloneData.artifactTypeProvides["artifact_group"]; ok {
+			standaloneData.artifactGroup = grp.(string)
+			delete(standaloneData.artifactTypeProvides, "artifact_group")
+		}
+	}
+	depends, err := installer.GetArtifactDepends()
+	if err != nil {
+		return nil, err
+	} else if depends != nil {
+		currentProvides, err := datastore.LoadProvidesFromStore(device.Store)
```

Comment:
Suggestion: In general the `FromStore` can be dropped, when the package name already signifies what is happening. Thus `datastore.LoadProvides` should be plenty I think :)